### PR TITLE
feature(docs): switch component theming

### LINKED ISSUE #1090: Theming: Switch
### Subject

Switch

### Description

Create a basic theming documentation for the `Switch` component that showcases how you can style it.

Theming file: [Switch theming file](https://github.com/chakra-ui/chakra-ui-docs/blob/main/content/docs/components/switch/theming.mdx)

### Template Reference

For single part components, feel free to use the `Button` component theming doc as a template reference: [Button theming](https://github.com/chakra-ui/chakra-ui-docs/blob/main/content/docs/components/button/theming.mdx).

For multipart components, feel free to use the `Input` component theming doc as a template reference: [Input theming](https://github.com/chakra-ui/chakra-ui-docs/blob/main/content/docs/components/input/theming.mdx).

### PR DESCRIPTION
**Pull request by kelvinwelter**
Closes #1090 

## 📝 Description

Adding theming docs for the `Switch` component

## 📝 Additional Information

I have some troubles when trying to document how to add custom sizes to the `Switch` component, so I didn't include this section in this PR.

### CODE REVIEW

File: content/docs/components/switch/theming.mdx [hunk 1]
```diff
@@ -2,3 +2,157 @@
...
+
+> To learn more about styling multipart components, visit the
+> [Component Style](/docs/styled-system/component-style#styling-multipart-components)
+> page.
+
+## Anatomy
+
+- A: `container`
+- B: `thumb`
+- C: `track`
```
Comment (nikolovlazar) [inline content/docs/components/switch/theming.mdx:17]: I've prepared a custom anatomy illustration for you. Place [this code](https://gist.github.com/nikolovlazar/28bda596bbb1fa8a2ec42f097c4ff3e0) below the anatomy list.

File: content/docs/components/switch/theming.mdx [hunk 2]
```diff
@@ -2,3 +2,157 @@
...
+  container: {
+      // ...
+  },
+  thumb: {
+      bg: "red.500",
+  },
+  track: {
+      bg: "gray.100",
+      _checked: {
+      bg: "gray.100",
```
Comment (nikolovlazar) [inline content/docs/components/switch/theming.mdx:63]: Missing indentation here.

File: content/docs/components/switch/theming.mdx [hunk 3]
```diff
@@ -2,3 +2,157 @@
...
+
+export const switchTheme = defineMultiStyleConfig({ baseStyle })
+```
+
+After customizing the switch theme, we can import it in our theme file and add
+it in the `components` property:
+
+```jsx live=false
+import { extendTheme } from '@chakra-ui/react'
+import { switchTheme } from './components/switch.ts'
```
Comment (nikolovlazar) [inline content/docs/components/switch/theming.mdx:76]: Remove the `.ts` extension from the import.

File: content/docs/components/switch/theming.mdx [hunk 4]
```diff
@@ -2,3 +2,157 @@
...
+})
+```
+
+
+Then, we can use the custom color scale as the color scheme for the button:
+
+```jsx live=false
+<Switch colorScheme='brand' />
+```
+
```
Comment (nikolovlazar) [inline content/docs/components/switch/theming.mdx:113]: You should provide an example on how to create a custom variant as well. Here's something to get you started: https://codesandbox.io/s/boxy-switch-tywwit?file=/src/index.tsx

File: content/docs/components/switch/theming.mdx [hunk 5]
```diff
@@ -2,3 +2,157 @@
...
+export const theme = extendTheme({
+  components: { Switch: switchTheme },
+})
+```
+
+> This is a crucial step to make sure that any changes that we make to the
+> switch theme are applied.
+
+
+## Using a custom color scheme
```
Comment (nikolovlazar) [inline content/docs/components/switch/theming.mdx:87]: You should put this block at the end of this section:

```markdown
Every time you're adding anything new to the theme, you'd need to run the CLI
command to get proper autocomplete in your IDE. You can learn more about the CLI
tool [here](/docs/styled-system/cli).
```

**Review by nikolovlazar — CHANGES_REQUESTED** [review]
A few more changes + the anatomy illustration and we're good to go.

File: content/docs/components/switch/theming.mdx (hunk 2 above)
Comment (kelvinwelter) [reply content/docs/components/switch/theming.mdx]: Fixed it on https://github.com/chakra-ui/chakra-ui-docs/pull/1117/commits/93f82edc931ac30fdb218e757b6d6eae003f0503

File: content/docs/components/switch/theming.mdx (hunk 3 above)
Comment (kelvinwelter) [reply content/docs/components/switch/theming.mdx]: Fixed it on https://github.com/chakra-ui/chakra-ui-docs/pull/1117/commits/611af6cae98f59d3bd19817174e3eff2d406ecbd

File: content/docs/components/switch/theming.mdx (hunk 5 above)
Comment (kelvinwelter) [reply content/docs/components/switch/theming.mdx:175]: Added it on https://github.com/chakra-ui/chakra-ui-docs/pull/1117/commits/e7d400ade4e8bd204d03648c977d8914a582dc9f

File: content/docs/components/switch/theming.mdx (hunk 1 above)
Comment (kelvinwelter) [reply content/docs/components/switch/theming.mdx:17]: Added it on https://github.com/chakra-ui/chakra-ui-docs/pull/1117/commits/9acf3dc59cea08fe18bd838d3c84b2dd0bf32bea

File: content/docs/components/switch/theming.mdx (hunk 4 above)
Comment (kelvinwelter) [reply content/docs/components/switch/theming.mdx:201]: Added it on https://github.com/chakra-ui/chakra-ui-docs/pull/1117/commits/d656e36236d45f73fa33b9240bc721b577e571e1 and https://github.com/chakra-ui/chakra-ui-docs/pull/1117/commits/e7a26b2632f4273aea12f37a631e274692a38d45 

Thank you very much for the example you provided!

**Review by nikolovlazar — APPROVED** [review]

**Comment by nikolovlazar** [comment]
Looking good! Thanks for contributing to the switch theming docs @kelvinwelter!

**Comment by nikolovlazar** [comment]
@allcontributors please add @kelvinwelter for docs.

**Pull request merged**